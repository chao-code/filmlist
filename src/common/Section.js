import PropTypes from 'prop-types'

const Section = ({ name, children, className, large }) => (
  <section className={className}>
    {large ? (
      <h3 className="mb-0">{name}</h3>
    ) : (
      <h6 className="mb-0">{name}</h6>
    )}
    <hr className="mt-1" />
    {children}
  </section>
)

Section.propTypes = {
  name: PropTypes.string.isRequired,
  children: PropTypes.node,
  className: PropTypes.string,
  large: PropTypes.bool,
}

Section.defaultProps = {
  children: null,
  className: '',
  large: false,
}

export default Section
