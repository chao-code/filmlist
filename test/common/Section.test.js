import { shallow } from 'enzyme'
import Section from '../../src/common/Section'

describe('<Section />', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<Section name="Name">Children</Section>)
    expect(wrapper).toMatchSnapshot()
  })
})
