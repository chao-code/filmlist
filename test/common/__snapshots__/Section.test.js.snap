// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Section /> should render correctly 1`] = `
<section
  className=""
>
  <h6
    className="mb-0"
  >
    Name
  </h6>
  <hr
    className="mt-1"
  />
  Children
</section>
`;
